src/EventListener/ProductPublisher.php

```diff
@@ -54,5 +95,47 @@ public function postFlush(PostFlushEventArgs $event)
         }
 
         $this->scheduledProducts = [];
+
+        foreach ($this->scheduledUpdates as $product) {
+            $this->eventBus->handle(ProductUpdated::occur($product));
+        }
+
+        $this->scheduledUpdates = [];
+
+        foreach ($this->scheduledDeletions as $product) {
+            $this->eventBus->handle(ProductDeleted::occur($product));
+        }
+
+        $this->scheduledDeletions = [];
+    }
+
+    /**
+     * @param $entity
+     *
+     * @return ProductInterface|\Sylius\Component\Product\Model\ProductInterface|\Sylius\Component\Resource\Model\TranslatableInterface
+     */
+    protected function getProductFromEntity($entity): ?ProductInterface
+    {
+        if ($entity instanceof ProductVariantInterface) {
+            $entity = $entity->getProduct();
+        } elseif ($entity instanceof ProductVariantTranslation) {
+            $entity = $entity->getTranslatable()->getProduct();
+        } elseif ($entity instanceof ProductTranslation) {
+            $entity = $entity->getTranslatable();
+        } elseif ($entity instanceof ChannelPricingInterface) {
+            $entity = $entity->getProductVariant()->getProduct();
+        } elseif ($entity instanceof ChannelPricingInterface) {
+            $entity = $entity->getProductVariant()->getProduct();
+        } elseif ($entity instanceof ProductTaxon) {
+            $entity = $entity->getProduct();
+        } elseif ($entity instanceof ProductAttributeValueInterface) {
+            $entity = $entity->getProduct();
+        } elseif ($entity instanceof ProductInterface) {
+            return $entity;
```

Comment:
I think we also need the check on ProductImageInterface

Reply:
Good catch, i'm gonna implement and test it today